plume-models/src/posts.rs

```diff
@@ -181,11 +181,14 @@ impl Post {
     }
 
     pub fn to_json(&self, conn: &PgConnection) -> serde_json::Value {
+        let blog = self.get_blog(conn);
         json!({
             "post": self,
             "author": self.get_authors(conn)[0].to_json(conn),
-            "url": format!("/~/{}/{}/", self.get_blog(conn).actor_id, self.slug),
-            "date": self.creation_date.timestamp()
+            "url": format!("/~/{}/{}/", blog.actor_id, self.slug),
+            "date": self.creation_date.timestamp(),
+            "blog": blog,
+            "blog_url": format!("/~/{}", blog.actor_id)
```

Comment:
This won't work with federated blogs, since `actor_id` only contains the part before the `@`. The best solution would be be to use `blog.to_json` for the `"blog"` field above, and then use the `fqn` field in the view to build the URL.

(BTW, I'm seeing that the `url` field is wrong too, but that's another issue)

Reply:
The problem is that Blog doesn't have an implementation of to_json :/ (and if I try using the macro `json!`, it works but I don't have the field fqn)

Reply:
Ok, your solution looks good then. And we can always add the `Blog::to_json` function in the future to make things simpler.

Reply:
Yep sure ^^
Then, how can I get the url of the corresponding instance? As I don't have access to a fqn property

Reply:
I can't also take some time to write a basic method for `Blog::to_json` if you wish ^^